pkg/query/api/v1.go

```diff
@@ -550,7 +544,7 @@ func (api *API) series(r *http.Request) (interface{}, []error, *ApiError) {
 	if set.Err() != nil {
 		return nil, nil, &ApiError{errorExec, set.Err()}
 	}
-	return metrics, warnings, nil
+	return metrics, set.Warnings(), nil
```

Comment:
it looks like nothing in this function returns a single warning anymore; why not change the function signature to eliminate this extra returned value?

Reply:
How do you mean? Because of the flow changes warnings returned by the set. And we want to return those warnings to the user, so we have those in the handler function signature. I don't quite get it. Could you elaborate on this?

Reply:
ah, nevermind, I thought we were always returning an empty list of warning here, but I see we are generating the warnings from the set